fe/be-java-extensions/jdbc-scanner/src/main/java/org/apache/doris/jdbc/BaseJdbcExecutor.java

```diff
@@ -207,32 +208,68 @@ public long getBlockAddress(int batchSize, Map<String, String> outputParams) thr
 
             if (isNullableString == null || replaceString == null) {
                 throw new IllegalArgumentException(
-                        "Output parameters 'is_nullable' and 'replace_string' are required.");
+                    "Output parameters 'is_nullable' and 'replace_string' are required.");
             }
 
             String[] nullableList = isNullableString.split(",");
             String[] replaceStringList = replaceString.split(",");
             curBlockRows = 0;
-            int columnCount = resultSetMetaData.getColumnCount();
 
-            initializeBlock(columnCount, replaceStringList, batchSize, outputTable);
+            int outputColumnCount = outputTable.getColumns().length;
+            initializeBlock(outputColumnCount, replaceStringList, batchSize, outputTable);
+
+            Map<String, Integer> resultSetColumnMap = new HashMap<>();
+            int resultSetColumnCount = resultSetMetaData.getColumnCount();
+            for (int i = 1; i <= resultSetColumnCount; i++) {
+                String columnName = resultSetMetaData.getColumnName(i).trim().toLowerCase();
+                resultSetColumnMap.put(columnName, i);
+            }
+
+            Map<String, Integer> columnIndexMap = new HashMap<>();
+            for (int j = 0; j < outputColumnCount; j++) {
+                String outputColumnName = outputTable.getFields()[j].trim().toLowerCase();
+                Integer resultSetIndex = resultSetColumnMap.get(outputColumnName);
+                if (resultSetIndex == null) {
+                    throw new RuntimeException("Column not found: " + outputColumnName);
+                }
+                columnIndexMap.put(outputColumnName, resultSetIndex - 1);
```

Comment:
why `resultSetIndex - 1`, add comment in code to explain.
Better give an example